modules/apps/29-fee/ibc_middleware.go

```diff
@@ -347,3 +350,14 @@ func (im IBCMiddleware) WriteAcknowledgement(
 func (im IBCMiddleware) GetAppVersion(ctx sdk.Context, portID, channelID string) (string, bool) {
 	return im.keeper.GetAppVersion(ctx, portID, channelID)
 }
+
+// UnmarshalPacketData attempts to use the underlying app to unmarshal the packet data.
+// If the underlying app does not support the PacketDataUnmarshaler interface, an error is returned.
+func (im IBCMiddleware) UnmarshalPacketData(bz []byte) (interface{}, error) {
```

Comment:
this is only necessary if the ADR 8 implementation is higher than ICS 29, which should not be the case. I added the function anyways so we could discuss here whether to keep it or not

Reply:
hmm, I guess its best to keep it?

Reply:
Agree that it should be kept. Let's remove ordering requirements from our stack whenever possible